### PR TITLE
Instant Summons will now brutally rip out embedded objects, and drag the embedee to you if nearby.

### PR DESCRIPTION
**Pull request by Rhials**
## About The Pull Request

The instant summons spell will cause embedded weapons to be violently torn out from victims upon being recalled away. This comes with bonus removal damage and a piercing wound. If you are near the victim as this happens, they will be violently launched towards you. Make sure not to let them fly into you!

Feel free to try it with the High-Frequency Blade (100% embed chance) next time you're wizard.

Also, if the numbers on this seem a bit extreme, give me a heads up. Combat balance isn't exactly my forte.

https://user-images.githubusercontent.com/28870487/206542566-0cf72e21-332e-48a5-801b-0f8478ef24ff.mp4

## Why It's Good For The Game

Adds an interesting synergy to one of the more commonly accessible spells. Get creative, get violent.

Encourages people to use summons for more interesting things than shove/steal insurance.

## Changelog

:cl:
balance: Instant Summons will now violently tear out embedded objects as they are summoned, and will drag embedee to you if nearby.
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
GET OVER HERE

**Comment by Serxule** [comment]
oh god... edagger wizard

**Comment by Maxipat112** [comment]
It should have a ripping sound

**Comment by Serxule** [comment]
> It should have a ripping sound

agree, the sound it makes is the opposite of brutal

**Comment by KittyNoodle** [comment]
throwing star yoyo when??

File: code/datums/components/embedded.dm [hunk 1]
```diff
@@ -287,3 +288,28 @@
 	to_chat(user, span_notice("You successfully pluck [weapon] from [victim]'s [limb.plaintext_zone]."))
 	to_chat(victim, span_notice("[user] plucks [weapon] from your [limb.plaintext_zone]."))
 	safeRemove(user)
+
+/// Called when an object is ripped out of someone's body by magic or other abnormal means
+/datum/component/embedded/proc/magic_pull(datum/source, mob/living/caster, obj/marked_item)
+	SIGNAL_HANDLER
+
+	if(marked_item != weapon || src.limb != limb)
```
Comment (JohnFulpWillard) [inline code/datums/components/embedded.dm:296]: ``limb`` isn't defined as anything so it's still src.limb, this is just checking if(limb != limb)

File: code/datums/components/embedded.dm [hunk 2]
```diff
@@ -287,3 +288,28 @@
...
+		var/dist = get_dist(caster, victim) //Check if the caster is close enough to yank them in
+		if(dist < 7)
+			victim.throw_at(caster, get_dist(victim, caster) - 1, 1, caster)
+			victim.Paralyze(1 SECONDS)
+			limb.force_wound_upwards(/datum/wound/pierce/moderate)
+			playsound(get_turf(victim), 'sound/effects/wounds/blood2.ogg', 50, TRUE)
+			victim.visible_message(span_alert("[victim] is sent flying towards [caster] as the [marked_item] tears out of them!"), span_alert("You are launched at [caster] as the [marked_item] tears from your body and towards their hand!"))
+		victim.visible_message(span_danger("[marked_item] is violently torn from [victim.name]'s [limb.plaintext_zone]!"), span_userdanger("[weapon] is violently torn from your [limb.plaintext_zone]!"))
+	else
+		victim.visible_message(span_danger("[marked_item] vanishes from [victim.name]'s [limb.plaintext_zone]!"), span_userdanger("[weapon] vanishes from [limb.plaintext_zone]!"))
```
Comment (JohnFulpWillard) [inline code/datums/components/embedded.dm:315]:
```suggestion
	if(!harmful)
		victim.visible_message(span_danger("[marked_item] vanishes from [victim.name]'s [limb.plaintext_zone]!"), span_userdanger("[weapon] vanishes from [limb.plaintext_zone]!"))
		return
	var/damage = weapon.w_class * remove_pain_mult
	limb.receive_damage(brute=(1-pain_stam_pct) * damage * 1.5, sharpness=SHARP_EDGED) // Performs exit wounds and flings the user to the caster if nearby
	victim.adjustStaminaLoss(pain_stam_pct * damage)

	var/dist = get_dist(caster, victim) //Check if the caster is close enough to yank them in
	if(dist < 7)
		victim.throw_at(caster, get_dist(victim, caster) - 1, 1, caster)
		victim.Paralyze(1 SECONDS)
		limb.force_wound_upwards(/datum/wound/pierce/moderate)
		playsound(get_turf(victim), 'sound/effects/wounds/blood2.ogg', 50, TRUE)
		victim.visible_message(span_alert("[victim] is sent flying towards [caster] as the [marked_item] tears out of them!"), span_alert("You are launched at [caster] as the [marked_item] tears from your body and towards their hand!"))
	victim.visible_message(span_danger("[marked_item] is violently torn from [victim.name]'s [limb.plaintext_zone]!"), span_userdanger("[weapon] is violently torn from your [limb.plaintext_zone]!"))
```

**Comment by Overuseddart** [comment]
Miners rejoice! Instant summons on miners finally useful.

**Comment by moocowswag** [comment]
Will actually be kind of disgusting with spear + instant summon as miner.

**Comment by Overuseddart** [comment]
> Will actually be kind of disgusting with spear + instant summon as miner and that will need to be monitored.

I will roleplay scorpion and you can’t stop me

**Comment by ju45he** [comment]
what if when you use instant summons on something glass that breaks and then embeds, when you summon it then it would reconstruct the glass

**Comment by Shadowflame909** [comment]
> Will actually be kind of disgusting with spear + instant summon as miner and that will need to be monitored.

yeah but think of how scary ash lizards who somehow survived multiple tendrils to make a bone spear and get a necropolis drop will be.

**Comment by Overuseddart** [comment]
> > Will actually be kind of disgusting with spear + instant summon as miner and that will need to be monitored.
> 
> yeah but think of how scary ash lizards who somehow survived multiple tendrils to make a bone spear and get a necropolis drop will be.

That depends on the ash walkers somehow not luring a megafauna to their base so I wouldn’t worry about it.

**Comment by Shadowflame909** [comment]
> miners once again proving why we shouldn't have given them wizard spells this is why we can't have nice things!

miners are fine with not having them if megafauna didn't have them either

unstoppable object meets immovable force!

**Pull request merged**